### PR TITLE
import_kvpb: change `WriteEngine` from a stream to a single message

### PR DESCRIPTION
**Pull request by lance6716**
This PR is to [solve tikv/importer#3](https://github.com/tikv/importer/issues/3) .

### CODE REVIEW

**Comment by kennytm** [comment]
You may need to manually edit the Rust source.

```
error[E0432]: unresolved imports `crate::import_kvpb::write_engine_request`, `crate::import_kvpb::WriteBatch`, `crate::import_kvpb::WriteEngineRequest`, `crate::import_kvpb::WriteHead`
  --> src/lib.rs:16:30
   |
16 |     use crate::import_kvpb::{write_engine_request, WriteBatch, WriteEngineRequest, WriteHead};
   |                              ^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^  ^^^^^^^^^ no `WriteHead` in `protos::import_kvpb`
   |                              |                     |           |
   |                              |                     |           no `WriteEngineRequest` in `protos::import_kvpb`
   |                              |                     no `WriteBatch` in `protos::import_kvpb`
   |                              no `write_engine_request` in `protos::import_kvpb`
help: a similar name exists in the module
   |
16 |     use crate::import_kvpb::{WriteEngineV3Request, WriteBatch, WriteEngineRequest, WriteHead};
   |                              ^^^^^^^^^^^^^^^^^^^^
help: a similar name exists in the module
   |
16 |     use crate::import_kvpb::{write_engine_request, WriteBatch, WriteEngineV3Request, WriteHead};
   |                                                                ^^^^^^^^^^^^^^^^^^^^
```

**Comment by lance6716** [comment]
> You may need to manually edit the Rust source.
> 
> ```
> error[E0432]: unresolved imports `crate::import_kvpb::write_engine_request`, `crate::import_kvpb::WriteBatch`, `crate::import_kvpb::WriteEngineRequest`, `crate::import_kvpb::WriteHead`
>   --> src/lib.rs:16:30
>    |
> 16 |     use crate::import_kvpb::{write_engine_request, WriteBatch, WriteEngineRequest, WriteHead};
>    |                              ^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^  ^^^^^^^^^ no `WriteHead` in `protos::import_kvpb`
>    |                              |                     |           |
>    |                              |                     |           no `WriteEngineRequest` in `protos::import_kvpb`
>    |                              |                     no `WriteBatch` in `protos::import_kvpb`
>    |                              no `write_engine_request` in `protos::import_kvpb`
> help: a similar name exists in the module
>    |
> 16 |     use crate::import_kvpb::{WriteEngineV3Request, WriteBatch, WriteEngineRequest, WriteHead};
>    |                              ^^^^^^^^^^^^^^^^^^^^
> help: a similar name exists in the module
>    |
> 16 |     use crate::import_kvpb::{write_engine_request, WriteBatch, WriteEngineV3Request, WriteHead};
>    |                                                                ^^^^^^^^^^^^^^^^^^^^
> ```

The rust source should keep synchronized with [importer PR#17](https://github.com/tikv/importer/pull/17), I wish to ignore this CI failed to break circular dependency. If this PR has some error, importer's PR will fail anyway.

**Comment by kennytm** [comment]
@lance6716 Importer depends on kvproto but not the other way around, there shouldn't be circular dependency

**Comment by lance6716** [comment]
> @lance6716 Importer depends on kvproto but not the other way around, there shouldn't be circular dependency

I'll solve that tomorrow 😭

**Comment by lance6716** [comment]
Ping @kennytm @overvenus

File: proto/import_kvpb.proto [hunk 1]
```diff
@@ -37,7 +37,7 @@ service ImportKV {
     // Open an engine.
     rpc OpenEngine(OpenEngineRequest) returns (OpenEngineResponse) {}
     // Open a write stream to the engine.
-    rpc WriteEngine(stream WriteEngineRequest) returns (WriteEngineResponse) {}
```
Comment (kennytm) [inline proto/import_kvpb.proto]: Could you keep this old method? This allows backward compatibility.

Comment (lance6716) [reply proto/import_kvpb.proto]: Oh Sure. New name indicates keeping backward compatibility, which I just realized :P

**Comment by lance6716** [comment]
> The *.proto part LGTM. Not sure if `src/lib.rs` really needs to be changed.
> 
> PTAL @BusyJay

since we should keep the old method, I reverted changes on `src/lib.rs`.

**Comment by lance6716** [comment]
@kennytm This repo needs 2 approving reviews, not 1 + 2 LGTM 😛

**Pull request merged**